packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts

```diff
@@ -244,12 +250,16 @@ export class Trail extends Resource {
         });
       }
 
-      logsRole = new iam.Role(this, 'LogsRole', { assumedBy: cloudTrailPrincipal });
+      if (props.cloudWatchLogRole) {
```

Comment:
i'm wondering if the behaviour of the property being ignored if `sendToCloudWatchLogs` is unset/disabled is the incorrect behaviour.

also curious: If a role/log group is provided wouldn't it be a better experience to set the `sendToCloudWatchLogs` to true as it's likely intended when those properties are set?

Reply:
> i'm wondering if the behaviour of the property being ignored if sendToCloudWatchLogs is unset/disabled is the incorrect behaviour.

Good point. A role could be assigned even if `sendToCloudWatchLogs` feature is not enabled.

> also curious: If a role/log group is provided wouldn't it be a better experience to set the sendToCloudWatchLogs to true as it's likely intended when those properties are set?

As a Developer I prefer explicit over implicit. If `sendToCloudWatchLogs` is set to `false` then it should be disabled. But If the value is undefined, it could be calculated.

Reply:
@AnderEnder we do set this default for `encryptionEnabled` to true encryption keys in numerous places.  example: Kinesis [stream encryption](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-kinesis/lib/stream.ts#L322-L324). The CDK should be as intent-based as possible when we can.

what's the behaviour when a role is provided but `sendToCloudWatchLogs` is unset? - wouldn't providing the role imply intent to send to CloudWatch? - perhaps i'm missing a use case where the alternative is intended.